src/function/gds/rec_joins.cpp

```diff
@@ -202,5 +202,22 @@ void RJAlgorithm::exec(processor::ExecutionContext* executionContext) {
     sharedState->mergeLocalTables();
 }
 
+bool SPAlgorithm::terminateFunc(GDSFrontier& frontier,
+    const processor::NodeOffsetMaskMap& maskMap) {
+    if (!maskMap.enabled()) {
+        return false;
+    }
+    for (auto& [tableID, mask] : maskMap.getMaskMap()) {
+        if (frontier.pinTableID(tableID)) {
+            for (auto i = 0u; i < mask->getMaxOffset(); ++i) {
+                if (mask->isMasked(i) && !frontier.isActive(i)) {
+                    return false;
```

Comment:
During initializing SPAlgorithm, you set numDstReached=0, or it's constructed that way.
numDstReached++;

Reply:
Done